### PR TITLE
Feature/solid

### PR DESCRIPTION
**Pull request by hienng99**

### CODE REVIEW

File: interfacesegregationprinciple/HinhChuNhat.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package solid.interfacesegregationprinciple;
+
+public class HinhChuNhat implements IMul {
```
Comment (QuangCuongIT) [inline interfacesegregationprinciple/HinhChuNhat.java]: Please use English only. Check the other places as well! Thx

File: interfacesegregationprinciple/HinhTron.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package solid.interfacesegregationprinciple;
```
Comment (QuangCuongIT) [inline interfacesegregationprinciple/HinhTron.java]: I think it should be interface.segregation.principle

File: liskovsubstitutionprinciple/Animal.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package solid.liskovsubstitutionprinciple;
```
Comment (QuangCuongIT) [inline liskovsubstitutionprinciple/Animal.java]: The same story with "interfacesegregationprinciple"

Comment (QuangCuongIT) [reply liskovsubstitutionprinciple/Animal.java]: Please check and change the others

**Comment by QuangCuongIT** [comment]
It's ok for now, but for the assignment. Please add the java docs for all methods and all classes. I will give you the format in the meeting.

File: interfacesegregationprinciple/HinhChuNhat.java (hunk 1 above)
Comment (QuangCuongIT) [reply interfacesegregationprinciple/HinhChuNhat.java]: Remove the "I" prefix for interface. For the our project standard, we do not use it

File: interfacesegregationprinciple/HinhTron.java (hunk 2 above)
Comment (hienng99) [reply interfacesegregationprinciple/HinhTron.java]: I renamed it to interfacesegregation.principle bc interface is not valid for package name.

File: interfacesegregationprinciple/HinhChuNhat.java (hunk 1 above)
Comment (hienng99) [reply interfacesegregationprinciple/HinhChuNhat.java]: I have adjusted.

File: liskovsubstitutionprinciple/Animal.java (hunk 3 above)
Comment (hienng99) [reply liskovsubstitutionprinciple/Animal.java]: I have adjusted

**Review by QuangCuongIT — APPROVED** [review]

File: interfacesegregation/principle/Circle.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+	
+	private int r;
+
+	@Override
+	public int pow(int a, int b) {
+		//Implement method
+		return 1;
+	}
+	
+	public double tinhDienTich() {
```
Comment (QuangCuongIT) [inline interfacesegregation/principle/Circle.java]: please use English

File: dependency/inversion/principle/DBConnector.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+	@Override
+	public void save(Object obj) {
+
+		//save the object toMySQL
+		
+	}
+	
+	
+
```
Comment (QuangCuongIT) [inline dependency/inversion/principle/DBConnector.java:11]: Please remove the unused break line. Check the other as well

File: openandclosed/principle/Rectangle.java [hunk 6]
```diff
@@ -0,0 +1,12 @@
+package solid.openandclosed.principle;
+
+public class Rectangle implements Polygons{
+	
+	private double canh;
```
Comment (QuangCuongIT) [inline openandclosed/principle/Rectangle.java]: Vietnamese again

File: dependency/inversion/principle/DBConnector.java (hunk 5 above)
Comment (hienng99) [reply dependency/inversion/principle/DBConnector.java:11]: I have adjusted

**Pull request merged**